.github/workflows/build_deploy_rcb_be.yml

```diff
@@ -0,0 +1,143 @@
+name: Build & Deploy RecipeBox Backend
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - recipe-box-backend/**
+      - .github/workflows/build_deploy_rcb_be.yml
+  pull_request:
+    branches:
+      - main
+    paths:
+      - recipe-box-backend/**
+      - .github/workflows/build_deploy_rcb_be.yml
+
+# Azure login
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Download and setup Java
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: '21'
+
+      - name: Download and install maven
+        run: sudo apt -qq update & sudo apt -qq install maven -y
+
+      - name: Package server app using maven
+        run: mvn package -B -q -DskipTests=true # TODO: remove skipping tests
+        working-directory: recipe-box-backend
+
+      - name: Upload generated bundle as an artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: production-rcb-be
+          path: ./recipe-box-backend/target/recipe-box-backend-0.0.1-SNAPSHOT.jar # TODO: adapt version / ask team to use finalName
+
+  deploy:
+    runs-on: ubuntu-latest
+    if: ${{ github.event_name == 'push' }}
```

Comment:
O sa vrem sa facem deploy doar daca este workflow dispatch sau este push in main